app/controllers/v1/versions_controller.rb

```diff
@@ -0,0 +1,24 @@
+module V1
+  class VersionsController < ApplicationController
+    include Authenticated
+
+    before_action :check_auth_token
+    before_action :find_purl
+
+    rescue_from VersionedFilesService::BadRequestError do |e|
+      render build_error('400', e, 'Bad request')
+    end
+
+    def withdraw
+      return render json: { error: "already deleted" }, status: :conflict if @purl.deleted?
+
+      PurlAndStacksService.withdraw(purl: @purl, version: params[:version], withdrawn: params[:withdrawn])
```

Comment:
Not seeing a render when this succeeds. Is the default controller behavior here to return an empty 204 response?

Reply:
Yes.